server/src/main/java/org/elasticsearch/action/admin/indices/rollover/RolloverRequest.java

```diff
@@ -148,76 +163,69 @@ public void dryRun(boolean dryRun) {
      * Adds condition to check if the index is at least <code>age</code> old
      */
     public void addMaxIndexAgeCondition(TimeValue age) {
-        this.conditions.add(new MaxAgeCondition(age));
+        if (this.conditions.add(new MaxAgeCondition(age)) == false) {
+            throw new IllegalArgumentException(MaxAgeCondition.NAME + " condition is already set");
+        }
     }
 
     /**
      * Adds condition to check if the index has at least <code>numDocs</code>
      */
     public void addMaxIndexDocsCondition(long numDocs) {
-        this.conditions.add(new MaxDocsCondition(numDocs));
+        if (this.conditions.add(new MaxDocsCondition(numDocs)) == false) {
+            throw new IllegalArgumentException(MaxDocsCondition.NAME + " condition is already set");
+        }
     }
 
     /**
      * Adds a size-based condition to check if the index size is at least <code>size</code>.
      */
     public void addMaxIndexSizeCondition(ByteSizeValue size) {
-        this.conditions.add(new MaxSizeCondition(size));
+        if (this.conditions.add(new MaxSizeCondition(size)) == false) {
+            throw new IllegalArgumentException(MaxSizeCondition.NAME + " condition is already set");
+        }
     }
 
-    /**
-     * Sets rollover index creation request to override index settings when
-     * the rolled over index has to be created
-     */
-    public void setCreateIndexRequest(CreateIndexRequest createIndexRequest) {
```

Comment:
it was used but I changed a bit the way this request works. The inner create index request is always set at initialization time, hence it can never be null. Its index name is never read though, but what matters is the mappings, settings and aliases set to it. One can easily do request.getCreateIndexRequest().mappings(...) etc. rather than replacing the whole create index request I thought, so I have a slight preference for retrieving the existing create index request and modify it rather than setting a new one. There is a single way to set mappings, settings and aliases this way. I find it clearer, although not perfect :)